tests/pipeline_runs_test.go

```diff
@@ -0,0 +1,76 @@
+//go:build test_integration
+
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	TestUtil "github.com/opendatahub-io/data-science-pipelines-operator/tests/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (suite *IntegrationTestSuite) TestPipelineSuccessfulRun() {
+
+	podCount := 7
+	suite.T().Run("Should create a Pipeline Run", func(t *testing.T) {
+		// Retrieve Pipeline ID to create a new run
+		pipelineID := TestUtil.RetrievePipelineId(t, APIServerURL)
+		postUrl := fmt.Sprintf("%s/apis/v2beta1/runs", APIServerURL)
+		body := TestUtil.FormatRequestBody(t, pipelineID)
+		contentType := "application/json"
+		// Create a new run
+		response, err := http.Post(postUrl, contentType, bytes.NewReader(body))
+		require.NoError(t, err)
+		responseData, err := ioutil.ReadAll(response.Body)
+		responseString := string(responseData)
+		loggr.Info(responseString)
+		require.NoError(t, err)
+		assert.Equal(t, 200, response.StatusCode)
+	})
+	// Waiting for the Pipeline run pods to be created
+	time.Sleep(5 * time.Minute)
+
+	suite.T().Run("Should successfully complete the Pipeleine Run", func(t *testing.T) {
+		t.Run(fmt.Sprintf("should have %d pods", podCount), func(t *testing.T) {
+			podList := &corev1.PodList{}
+			listOpts := []client.ListOption{
+				client.InNamespace(suite.DSPANamespace),
+			}
+			err := suite.Clientmgr.k8sClient.List(suite.Ctx, podList, listOpts...)
+			require.NoError(t, err)
+			// Filtering the pods created for iris-training-pipeline
+			filteredPods := []corev1.Pod{}
+			for _, pod := range podList.Items {
+				if strings.HasPrefix(pod.Name, "iris-training-pipeline") {
+					filteredPods = append(filteredPods, pod)
+				}
+			}
+			assert.Equal(t, podCount, len(filteredPods))
+		})
+	})
+}
```

Comment:
this is counting the number of pods that were created, this doesn't really test a successful run

check the kfp v2 api reference for pipelines, and retrieve the status field via api and confirm that it ended with a success state